### ISSUE TITLE
implement token refreshing for client credentials grant

### ISSUE THREAD

**Issue by spreadnuts** [circleguard/ossapi#21]
Sorry for the bother if this is an easy fix or I'm doing something wrong, I'm a bit new to this..

I have:
```python
from ossapi import *
api = OssapiV2(client_id=8114 client_secret='xxxxxxxxxxxx')
```
endpoints were all working perfectly but after a few days it looks like the session expires:

```
Traceback (most recent call last):
  File "C:\Users\Mathew\Desktop\Mathew\Programs\mouseonlyrecords\main.py", line 69, in <module>
    user_scores = api.user_scores(user_id=p[0], type_="best", limit=51, offset=0)
  File "C:\Users\Mathew\Desktop\Mathew\Programs\mouseonlyrecords\venv\lib\site-packages\ossapi\ossapiv2.py", line 83, in wrapper
    return function(*args, **kwargs)
  File "C:\Users\Mathew\Desktop\Mathew\Programs\mouseonlyrecords\venv\lib\site-packages\ossapi\ossapiv2.py", line 681, in user_scores
    return self._get(List[Score], f"/users/{user_id}/scores/{type_.value}",
  File "C:\Users\Mathew\Desktop\Mathew\Programs\mouseonlyrecords\venv\lib\site-packages\ossapi\ossapiv2.py", line 203, in _get
    r = self.session.get(f"{self.BASE_URL}{url}", params=params)
  File "C:\Users\Mathew\Desktop\Mathew\Programs\mouseonlyrecords\venv\lib\site-packages\requests\sessions.py", line 555, in get
    return self.request('GET', url, **kwargs)
  File "C:\Users\Mathew\Desktop\Mathew\Programs\mouseonlyrecords\venv\lib\site-packages\requests_oauthlib\oauth2_session.py", line 477, in request
    url, headers, data = self._client.add_token(
  File "C:\Users\Mathew\Desktop\Mathew\Programs\mouseonlyrecords\venv\lib\site-packages\oauthlib\oauth2\rfc6749\clients\base.py", line 198, in add_token
    raise TokenExpiredError()
oauthlib.oauth2.rfc6749.errors.TokenExpiredError: (token_expired)
```

**Comment by tybug** [comment]
Whoops, I forgot I never implemented token refreshing for client credentials grant. For now you can call `OssapiV2.clear_authentication()` before instantiating `OssapiV2` to wipe the expired token and force it to re-authenticate, but I will fix this properly at some point.

**Comment by tybug** [comment]
As an update, it doesn't seem like the api actually provides refresh tokens for the client credentials grant (see https://github.com/ppy/osu-web/issues/8023), so the solution here is just going to be re-authenticating whenever it expires. So the `OssapiV2.clear_authentication()` workaround above is actually a rather canonical solution for now (though I will of course implement this in the library itself soon).

**Comment by tybug** [comment]
closed by dbfb193
